### ISSUE TITLE
Update PSBoundParameters later to make sure we pick up all changes

### ISSUE THREAD

**Issue by heaths** [Azure/azure-sdk-tools#1177]
* We need to update PSBoundParameters here.
* And ProvisionerApplicationSecret and TenantId
* TestApplicationOid is also updated later in the script.
* BaseName is updated later as well.
* As well as ResourceGroupName

_Originally posted by @pakrym in https://github.com/Azure/azure-sdk-tools/pull/1175#r518910453_

**Comment by heaths** [comment]
@benbp I'll let you get your work in first so this doesn't conflict. This issue shouldn't regress anything since people clearly weren't using "newly" added parameters, or they'd have been broken before. And my change that needs the `$BaseName` isn't going in until 1) the Az module is updated on agents, or 2) we have a story for prereq'ing newer Az modules (e.g. az.keyvault@3.0.0).

/cc @weshaggard @pakrym

**Comment by pakrym** [comment]
@heaths, were you intending to close this?

**Comment by heaths** [comment]
Yes, I fixed it with the latest changes we sync'd. I still kept updating of `$PSBoundParameters` closer to where we set/overwrote the variables because I still think it's more tenable than trying to remember all the variables we need to overwrite in `$PSBoundParameters` later, but the intent of this bug was addressed (i.e. I missed at least `$BaseName`).

**Comment by pakrym** [comment]
I don't think so, there a few that are listed in the issue that are not updated.

**Comment by heaths** [comment]
I see: https://github.com/Azure/azure-sdk-tools/pull/1175/files#r518911288
